# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build util-macros-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.63; am-1.11; lt-2.2.6)
libtoolize -c -f; aclocal -I m4; autoheader; autoconf; automake -a -c
./configure --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/include
install -m 644 include/pciaccess.h $DESTDIR/x11/include
install -d $DESTDIR/x11/lib
install src/.libs/libpciaccess.so.0.* $DESTDIR/x11/lib/libpciaccess.so.0
ln -sf libpciaccess.so.0 $DESTDIR/x11/lib/libpciaccess.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 pciaccess.pc $DESTDIR/x11/lib/pkgconfig

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/libpciaccess-build.pc << EOF
Name: libpciaccess
Version: 0.11.0
Description: Generic PCI access library
Requires: $REQUIRES

devel=\\
/x11/include/pciaccess.h \\
/x11/lib/libpciaccess.so \\
/x11/lib/pkgconfig/pciaccess.pc

exec=\\
/x11/lib/libpciaccess.so.0
EOF

exit

# patch
--- libpciaccess-0.11.0.orig/src/linux_sysfs.c
+++ libpciaccess-0.11.0/src/linux_sysfs.c
@@ -40,6 +40,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <limits.h>
 #include <sys/mman.h>
 #include <dirent.h>
 #include <errno.h>
@@ -382,7 +383,7 @@
 
 
     while ( temp_size > 0 ) {
-	const ssize_t bytes = pread64( fd, data_bytes, temp_size, offset );
+	const ssize_t bytes = pread( fd, data_bytes, temp_size, offset );
 
 	/* If zero bytes were read, then we assume it's the end of the
 	 * config file.
@@ -440,7 +441,7 @@
 
 
     while ( temp_size > 0 ) {
-	const ssize_t bytes = pwrite64( fd, data_bytes, temp_size, offset );
+	const ssize_t bytes = pwrite( fd, data_bytes, temp_size, offset );
 
 	/* If zero bytes were written, then we assume it's the end of the
 	 * config file.

# source
https://x.org/releases/individual/lib/libpciaccess-0.11.0.tar.gz
